pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/UnsafeUtils.java

```diff
@@ -0,0 +1,262 @@
+package com.hedera.pbj.runtime.io.buffer;
+
+import java.lang.reflect.Field;
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.security.AccessController;
+import java.security.PrivilegedExceptionAction;
+
+/**
+ * This class is providing access to some of the unsafe functionality for
+ * directly allocated {@link ByteBuffer} objects.
+ * It provides functions to query the VM for supporting the general functionality
+ * and functions to get the memory offsetts of the {@link ByteBuffer} object, so a
+ * direct and fast access to the data can be performed.
+ * The implementation is inspired by the Google's Prrotobuf implementation.
+ */
+final class UnsafeUtils {
+  private static final sun.misc.Unsafe UNSAFE = getUnsafe();
+  private static final MemoryAccessor MEMORY_ACCESSOR = getMemoryAccessor();
+  private static final boolean HAS_UNSAFE_BYTEBUFFER_OPERATIONS =
+      supportsUnsafeByteBufferOperations();
+  private static final boolean HAS_UNSAFE_ARRAY_OPERATIONS = supportsUnsafeArrayOperations();
+  static final long BYTE_ARRAY_BASE_OFFSET = arrayBaseOffset(byte[].class);
+  // Micro-optimization: we can assume a scale of 1 and skip the multiply
+  // private static final long BYTE_ARRAY_INDEX_SCALE = 1;
+  private static final long BUFFER_ADDRESS_OFFSET = fieldOffset(bufferAddressField());
+
+  private UnsafeUtils() {}
+  static boolean hasUnsafeByteBufferOperations() {
+    return HAS_UNSAFE_BYTEBUFFER_OPERATIONS;
+  }
+
+  private static int arrayBaseOffset(Class<?> clazz) {
+    return HAS_UNSAFE_ARRAY_OPERATIONS ? MEMORY_ACCESSOR.arrayBaseOffset(clazz) : -1;
+  }
+  static void copyMemory(byte[] src, long srcIndex, long targetOffset, long length) {
+    MEMORY_ACCESSOR.copyMemory(src, srcIndex, targetOffset, length);
+  }
+  static void copyMemory(long srcOffset, byte[] target, long targetIndex, long length) {
```

Comment:
All methods should have java doc. The java doc should explain any understanding needed when using method.

Reply:
Added JavaDoc to all the methods.